diff --git a/app/core/gimpbacktrace-linux.c b/app/core/gimpbacktrace-linux.c
index e4166e3..b6f7db9 100644
--- a/app/core/gimpbacktrace-linux.c
+++ b/app/core/gimpbacktrace-linux.c
@@ -39,7 +39,6 @@
 #include <fcntl.h>
 #include <dirent.h>
 #include <signal.h>
-#include <execinfo.h>
 #include <dlfcn.h>
 #include <string.h>
 #include <stdio.h>
@@ -279,7 +278,7 @@ gimp_backtrace_signal_handler (gint signum)
 
           if (thread->tid == tid)
             {
-              thread->n_frames = backtrace ((gpointer *) thread->frames,
+              thread->n_frames = unw_backtrace ((gpointer *) thread->frames,
                                             MAX_N_FRAMES);
 
               g_atomic_int_dec_and_test (&handler_n_remaining_threads);
