From 1239cb7f151e021f166d993ada0780e869110394 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh?= <thomas@t-8ch.de>
Date: Tue, 8 Jun 2021 00:24:00 +0200
Subject: [PATCH 02/11] ao_pipewire: whitespace fixes

---
 audio/out/ao_pipewire.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/audio/out/ao_pipewire.c b/audio/out/ao_pipewire.c
index 79965cdc68..b0102bee8e 100644
--- a/audio/out/ao_pipewire.c
+++ b/audio/out/ao_pipewire.c
@@ -121,9 +121,9 @@ static void on_param_changed(void *userdata, uint32_t id, const struct spa_pod *
 {
     struct ao *ao = userdata;
     struct priv *p = ao->priv;
-	const struct spa_pod *params[1];
-	uint8_t buffer[1024];
-	struct spa_pod_builder b = SPA_POD_BUILDER_INIT(buffer, sizeof(buffer));
+    const struct spa_pod *params[1];
+    uint8_t buffer[1024];
+    struct spa_pod_builder b = SPA_POD_BUILDER_INIT(buffer, sizeof(buffer));
 
     if (param == NULL || id != SPA_PARAM_Format)
         return;
-- 
2.34.1

