From 1f08d8507e87eb79e864a62303e10be873d0cd4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh?= <thomas@t-8ch.de>
Date: Fri, 10 Dec 2021 21:29:16 +0100
Subject: [PATCH 06/11] ao_pipewire: add meson support

---
 meson.build       | 8 ++++++++
 meson_options.txt | 1 +
 2 files changed, 9 insertions(+)

diff --git a/meson.build b/meson.build
index 4f4d66335d..955cc130d9 100644
--- a/meson.build
+++ b/meson.build
@@ -894,6 +894,13 @@ if oss
     sources += files('audio/out/ao_oss.c')
 endif
 
+pipewire = dependency('libpipewire-0.3', version: '>= 0.3', required: get_option('pipewire'))
+if pipewire.found()
+    dependencies += pipewire
+    features += 'pipewire'
+    sources += files('audio/out/ao_pipewire.c')
+endif
+
 pulse = dependency('libpulse', version: '>= 1.0', required: get_option('pulse'))
 if pulse.found()
     dependencies += pulse
@@ -1768,6 +1775,7 @@ conf_data.set10('HAVE_OPENAL', openal.found())
 conf_data.set10('HAVE_OPENSLES', opensles.found())
 conf_data.set10('HAVE_OSS_AUDIO', oss)
 conf_data.set10('HAVE_OSX_THREAD_NAME', osx_thread_name)
+conf_data.set10('HAVE_PIPEWIRE', pipewire.found())
 conf_data.set10('HAVE_POSIX', posix)
 conf_data.set10('HAVE_PULSE', pulse.found())
 conf_data.set10('HAVE_RPI', rpi['use'])
diff --git a/meson_options.txt b/meson_options.txt
index 196119e4d6..8313f5d716 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -47,6 +47,7 @@ option('jack', type: 'feature', value: 'auto', description: 'JACK audio output')
 option('openal', type: 'feature', value: 'disabled', description: 'OpenAL audio output')
 option('opensles', type: 'feature', value: 'auto', description: 'OpenSL ES audio output')
 option('oss-audio', type: 'feature', value: 'auto', description: 'OSSv4 audio output')
+option('pipewire', type: 'feature', value: 'auto', description: 'PipeWire audio output')
 option('pulse', type: 'feature', value: 'auto', description: 'PulseAudio audio output')
 option('sdl2-audio', type: 'feature', value: 'auto', description: 'SDL2 audio output')
 option('wasapi', type: 'feature', value: 'auto', description: 'WASAPI audio output')
-- 
2.34.1

