From 72723d360585470405d9d7fd696c11ed6dfd1053 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh?= <thomas@t-8ch.de>
Date: Sun, 12 Dec 2021 21:47:19 +0100
Subject: [PATCH 08/11] ao_pipewire: simplify option parsing

---
 audio/out/ao_pipewire.c | 30 +++++++++---------------------
 1 file changed, 9 insertions(+), 21 deletions(-)

diff --git a/audio/out/ao_pipewire.c b/audio/out/ao_pipewire.c
index a5c4322eeb..9d2b3a7423 100644
--- a/audio/out/ao_pipewire.c
+++ b/audio/out/ao_pipewire.c
@@ -14,27 +14,11 @@
 #include "internal.h"
 #include "osdep/timer.h"
 
-struct ao_pipewire_opts {
-    int buffer_samples;
-};
-
-#define OPT_BASE_STRUCT struct ao_pipewire_opts
-static const struct m_sub_options ao_pipewire_conf = {
-    .opts = (const struct m_option[]) {
-        {"pipewire-buffer-samples", OPT_INT(buffer_samples)},
-        {0}
-    },
-    .defaults = &(const struct ao_pipewire_opts) {
-        .buffer_samples = 2048,
-    },
-    .size = sizeof(struct ao_pipewire_opts),
-};
-
 struct priv {
     struct pw_thread_loop *loop;
     struct pw_stream *stream;
 
-    struct ao_pipewire_opts *opts;
+    int buffer_samples;
     bool muted;
     float volume[2];
 };
@@ -217,8 +201,6 @@ static int init(struct ao *ao)
     const struct spa_pod *params[1];
     char latency_str[64];
 
-    p->opts = mp_get_config_group(ao, ao->global, &ao_pipewire_conf);
-
     enum spa_audio_format spa_format = af_fmt_to_pw(ao->format);
     if (spa_format == SPA_AUDIO_FORMAT_UNKNOWN) {
         ao->format = AF_FORMAT_FLOATP;
@@ -244,7 +226,7 @@ static int init(struct ao *ao)
         ao->sstride = ao->channels.num * af_fmt_to_bytes(ao->format);
     }
 
-    ao->device_buffer = p->opts->buffer_samples;
+    ao->device_buffer = p->buffer_samples;
     snprintf(latency_str, sizeof(latency_str), "%d/%d", ao->device_buffer, ao->samplerate);
 
     pw_init(NULL, NULL);
@@ -361,6 +343,7 @@ static int control(struct ao *ao, enum aocontrol cmd, void *arg)
     }
 }
 
+#define OPT_BASE_STRUCT struct priv
 
 const struct ao_driver audio_out_pipewire = {
     .description = "PipeWire audio output",
@@ -378,6 +361,11 @@ const struct ao_driver audio_out_pipewire = {
     {
         .loop = NULL,
         .stream = NULL,
+        .buffer_samples = 2048
+    },
+    .options_prefix = "pipewire",
+    .options = (const struct m_option[]) {
+        {"buffer-samples", OPT_INT(buffer_samples)},
+        {0}
     },
-    .global_opts = &ao_pipewire_conf,
 };
-- 
2.34.1

