diff --git a/wscript b/wscript
index a0227e648e..e82444fef6 100644
--- a/wscript
+++ b/wscript
@@ -643,7 +643,7 @@ video_output_features = [
     }, {
         'name': '--vaapi-wayland',
         'desc': 'VAAPI (Wayland support)',
-        'deps': 'vaapi && gl-wayland',
+        'deps': 'vaapi && wayland',
         'func': check_pkg_config('libva-wayland', '>= 1.1.0'),
     }, {
         'name': '--vaapi-drm',
@@ -655,10 +655,15 @@ video_output_features = [
         'desc': 'VAAPI EGL on X11',
         'deps': 'vaapi-x11 && egl-x11',
         'func': check_true,
+    }, {
+        'name': '--vaapi-wayland-egl',
+        'desc': 'VAAPI EGL on Wayland',
+        'deps': 'vaapi-wayland && gl-wayland',
+        'func': check_true,
     }, {
         'name': 'vaapi-egl',
         'desc': 'VAAPI EGL',
-        'deps': 'vaapi-x-egl || vaapi-wayland || vaapi-drm',
+        'deps': 'vaapi-x-egl || vaapi-wayland-egl || vaapi-drm',
         'func': check_true,
     }, {
         'name': '--caca',
