diff --git a/lib/portage/util/env_update.py b/lib/portage/util/env_update.py.a
index bb0ebf8..bac5b6e 100644
--- a/lib/portage/util/env_update.py
+++ b/lib/portage/util/env_update.py.a
@@ -417,11 +417,6 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env, writemsg_lev
         systemd_gentoo_env.write(senvnotice)
 
         for env_key in env_keys:
-            # Skip PATH since this makes it impossible to use
-            # "systemctl --user import-environment PATH".
-            if env_key == "PATH":
-                continue
-
             env_key_value = env[env_key]
 
             # Skip variables with the empty string
