diff --git a/src/utils/Utils.vala b/src/utils/Utils.vala
index 7889d9d..6abb952 100644
--- a/src/utils/Utils.vala
+++ b/src/utils/Utils.vala
@@ -720,7 +720,7 @@ namespace GameHub.Utils
 		private static string color(ConsoleColor c, bool foreground)
 		{
 			var color_code = c + 30 + 60;
-			if(!foreground) color_code += 10;
+			if(!foreground) color_code = color_code + 10;
 			return "\x001b[%dm".printf(color_code);
 		}
 
