diff --git a/src/utils/NumberUtils.h b/src/utils/NumberUtils.h.a
index d0a7471..9308977 100644
--- a/src/utils/NumberUtils.h
+++ b/src/utils/NumberUtils.h.a
@@ -65,7 +65,7 @@ really_inline from_chars_result from_chars(const char *first, const char *last,
 #ifdef _WIN32
     tempval = _strtod_l(first, &endptr, loc.local);
 #else
-    tempval = ::strtod_l(first, &endptr, loc.local);
+    tempval = ::strtod(first, &endptr);
 #endif
 
     if (errno != 0)
@@ -104,7 +104,7 @@ really_inline from_chars_result from_chars(const char *first, const char *last,
     // On OSX, strtod_l is for some reason drastically faster than strtof_l.
     tempval = static_cast<float>(::strtod_l(first, &endptr, loc.local));
 #else
-    tempval = ::strtof_l(first, &endptr, loc.local);
+    tempval = ::strtof(first, &endptr);
 #endif
 
     if (errno != 0)
@@ -140,7 +140,7 @@ really_inline from_chars_result from_chars(const char *first, const char *last,
 #ifdef _WIN32
     tempval = _strtol_l(first, &endptr, 0, loc.local);
 #else
-    tempval = ::strtol_l(first, &endptr, 0, loc.local);
+    tempval = ::strtol(first, &endptr, 0);
 #endif
 
     if (errno != 0)
